.github/workflows/verify-devcontainers.yml

```diff
@@ -40,7 +46,45 @@ jobs:
         else
           echo "::note::Dev Container files are up-to-date."
         fi
+    - name: Inspect changes
+      if: ${{ inputs.base_sha != '' }}
+      id: inspect-changes
+      env:
+        BASE_SHA: ${{ inputs.base_sha }}
+      run: |
+        echo "Fetch history and determine merge base..."
+        git fetch origin --unshallow -q
+        git fetch origin $BASE_SHA -q
+        merge_base_sha=$(git merge-base $GITHUB_SHA $BASE_SHA)
+
+        echo "Head SHA: $GITHUB_SHA"
+        echo "PR Base SHA: $BASE_SHA"
+        echo "Merge Base SHA: $merge_base_sha"
+
+        echo "Checking for changes to devcontainer/matrix files..."
+
+        all_dirty_files=$(git diff --name-only "${merge_base_sha}" "${GITHUB_SHA}")
+        echo "::group::All dirty files"
+        echo "${all_dirty_files}"
+        echo "::endgroup::"
+
+        file_regex="^(.devcontainer|ci/matrix.yaml|.github/actions/workflow-build/build-workflow.py)"
+        echo "Regex: ${file_regex}"
+
+        relevant_dirty_files=$(echo "${all_dirty_files}" | grep -E "${file_regex}" || true)
+        echo "::group::Relevant dirty files"
+        echo "${relevant_dirty_files}"
+        echo "::endgroup::"
+
+        if [[ -z "${relevant_dirty_files}" ]]; then
+          echo "No relevant changes detected. Skipping devcontainer testing."
+          echo "skip=true" >> $GITHUB_OUTPUT
+        else
+          echo "Detected relevant changes. Continuing."
+          echo "skip=false" >> $GITHUB_OUTPUT
+        fi
```

Comment:
I [have used](https://github.com/rapidsai/devcontainers/blob/a7f4af11edb896a5b279be3c2a8be01cac31d695/.github/actions/image-matrix/action.yml#L31-L42) [tj-actions/changed-files](https://github.com/tj-actions/changed-files) to do this for a number of years. Has tons of options to control what gets compared (differences from prior commits on the PR branch, target branch, etc.), and the outputs are [super specific](https://github.com/rapidsai/devcontainers/actions/runs/9484855265/job/26135378198#step:3:73). Definitely worth looking at if you find yourself doing this more.